File: reload/AD/MR/Ex2.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Sep 23 02:28:58 2026

@author: jn
"""

import random

nSamples = 1000000
contA = 0
contB = 0

for i in range(nSamples):
    moedaRed = random.choices(["CA","CO"], weights = [2/5,3/5], k=1)[0]
    moedaGreen = random.choices(["CA","CO"], weights = [2/5,3/5], k=1)[0]
    moedaBlue = random.choices(["CA","CO"], weights = [2/5,3/5], k=1)[0]
    
    # (moedaRed == "CO" and moedaGreen == "CA") or (moedaRed == "CA" and moedaGreen == "CO")
    if (moedaRed != moedaGreen): 
        contA+=1
        
    if (moedaGreen == "CO" and moedaBlue == "CO"):
        contB+=1
        
PA = contA / nSamples        
PB = contB / nSamples
        
print("P(A):", PA)
print("P(B):", PB)

File: reload/AD/MR/Ex2.R

nSamples = 1000000
contA <- 0
contB <- 0

for (i in 1:nSamples)
{
  moedaRed = sample(c("CA","CO"), 1, prob=c(2/5,3/5))
  moedaGreen = sample(c("CA","CO"), 1, prob=c(2/5,3/5))
  moedaBlue = sample(c("CA","CO"), 1, prob=c(2/5,3/5))
  
  # (((moedaRed == "CA") & (moedaGreen == "CO")) | ((moedaRed == "CO") & (moedaGreen == "CA")))
  if (moedaRed != moedaGreen)
  {
    contA <- contA + 1
  }
  
  # if ((moedaGreen == "CO") & (MoedaBlue == "CO"))
  if ((moedaGreen == "CO") & (moedaBlue == moedaGreen))
  {
    contB <- contB + 1
  }
}

PA = contA / nSamples
PB = contB / nSamples

sprintf("P(A): %f", PA)
sprintf("P(B): %f", PB)
